pass-an-image.cpp

```diff
@@ -2,11 +2,11 @@
 #include "brightener.h"
 
 int main() {
-    Image image;
-    image.rows = 512;
-    image.columns = 512;
+    auto image = std::make_unique<Image>(512, 512);
 	std::cout << "Brightening a 512 x 512 image\n";
-    ImageBrightener brightener(image);
+
+    ImageBrightener brightener(std::move(image));
```

Comment:
`unique_ptr` is good for "automatic" memory management, in the sense that we don't have to do new and delete.
It also expresses ownership very clearly. In this case, it's enough to see the `main` function to know how the ownership goes... initially this function owns the image (due to `make_unique`). Then, it moves its responsibility (using `std::move`) to `ImageBrightener` constructor, which in turn moves it to the object (data-member). After moving, we cannot use `image` in `main` anymore, and `ImageBrightener` object is responsible to free that memory.
Without unique_ptr, we would need to see the whole implementation to search for the new and delete :)

**Question to ask**: Is it intuitive that `ImageBrightener` owns the brightened image? Because its job is only to brighten. Afterwards, I may want to display/process the image further and the ownership would need to move to them.